cmd/spire-server/cli/run/run.go

```diff
@@ -406,33 +427,36 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 		}
 
 		federatesWith := map[spiffeid.TrustDomain]bundleClient.TrustDomainConfig{}
-		for trustDomain, config := range c.Server.Federation.FederatesWith {
-			port := defaultBundleEndpointPort
-			if config.BundleEndpoint.Port != 0 {
-				port = config.BundleEndpoint.Port
-			}
-			if config.BundleEndpoint.UseWebPKI && config.BundleEndpoint.SpiffeID != "" {
-				return nil, errors.New("usage of `bundle_endpoint.spiffe_id` is not allowed when authenticating with Web PKI")
-			}
-
-			var spiffeID spiffeid.ID
-			if config.BundleEndpoint.SpiffeID != "" {
-				spiffeID, err = spiffeid.FromString(config.BundleEndpoint.SpiffeID)
-				if err != nil {
-					return nil, err
-				}
-			}
 
+		for trustDomain, config := range c.Server.Federation.FederatesWith {
 			td, err := spiffeid.TrustDomainFromString(trustDomain)
 			if err != nil {
 				return nil, err
 			}
 
-			federatesWith[td] = bundleClient.TrustDomainConfig{
-				EndpointAddress:  fmt.Sprintf("%s:%d", config.BundleEndpoint.Address, port),
-				EndpointSpiffeID: spiffeID,
-				UseWebPKI:        config.BundleEndpoint.UseWebPKI,
+			var trustDomainConfig *bundleClient.TrustDomainConfig
+			switch {
+			case config.DeprecatedBundleEndpoint != nil:
+				sc.Log.Warn("The `bundle_endpoint` configurable is deprecated and will be removed in a future release. Please use `bundle_endpoint_url` and `bundle_endpoint_profile` to configure the federation relationships instead.")
+				trustDomainConfig, err = parseDeprecatedBundleEndpoint(config.DeprecatedBundleEndpoint)
+				if err != nil {
+					return nil, fmt.Errorf("error parsing federation relationship for trust domain %q: %w", trustDomain, err)
+				}
+				trustDomainConfig.DeprecatedConfig = true
+				if spiffeAuth, ok := trustDomainConfig.EndpointProfile.(bundleClient.HTTPSSPIFFEProfile); ok {
+					if spiffeAuth.EndpointSPIFFEID.IsZero() {
+						sc.Log.Warnf("federation.federates_with[\"%s\"].bundle_endpoint.spiffe_id is not specified in the SPIFFE Authentication configuration. A specific SPIFFE ID will be required in a future release.", trustDomain)
```

Comment:
I think the CLI code would be a little easier to follow if we did this check inside bundle endpoint client instead, since we're using structs and methods from there.

Reply:
I remember evaluating that and the problem was that we don't have logging wired all the way to the bundle endpoint client. The code changes/complexity to add logging seemed to be more or less the same as having the check here so I finally decided to have it here. I think that it could be ok either way, what's your thought?

Reply:
Ok no problem :) after commenting I thought maybe it was something like that. I think it's ok for now ... we won't have to live with it for too long